src/daft-core/src/array/ops/hash.rs

```diff
@@ -186,6 +186,24 @@ impl FixedSizeListArray {
     }
 }
 
+impl StructArray {
+    pub fn hash(&self, seed: Option<&UInt64Array>) -> DaftResult<UInt64Array> {
+        // seed first child with input seed,
+        // then seed each child after with the output of the previous
+        if self.children.is_empty() {
+            return Err(DaftError::ValueError(
+                "Cannot hash struct with no children".into(),
+            ));
+        }
```

Comment:
hmm I see your point of hashing struct `a, b` and `b, a` what do you think about sorting the names and performing the same algorithm as below but in alphabetical order. 

I don't think it matters too much tbh

Reply:
Yeah, I'm also not too concerned